app/api/api_v1/endpoints/report.py

```diff
@@ -0,0 +1,143 @@
+            cult=utils.parse_generic_cultivation_info(json_file),
+            irri=utils.parse_irrigation(json_file),
+            fert=utils.parse_fertilization(json_file),
+            pdmd=utils.parse_plant_protection(json_file)
+        )
+
+    elif report_db.type == "plant-protection":
+
+        pdf = plant_protection(utils.parse_plant_protection(json.loads(report_db.file)))
+
+    elif report_db.type == "irrigations":
+
+        pdf = irrigations(utils.parse_irrigation(json.loads(report_db.file)))
+
+    elif report_db.type == "fertilisations":
+
+        pdf = fertilisation(utils.parse_fertilization(json.loads(report_db.file)))
+
+    elif report_db.type == "harvests":
+        # This will pass always, because it generates an empty .pdf every time
+        pdf = harvests()
+    else:
+        # Same as work-book for now
+        json_file = json.loads(report_db.file)
+
+        pdf = utils.work_book(
+            farm=utils.parse_farm_profile(json_file),
+            plot=utils.parse_plot_detail(json_file),
+            cult=utils.parse_generic_cultivation_info(json_file),
+            irri=utils.parse_irrigation(json_file),
+            fert=utils.parse_fertilization(json_file),
+            pdmd=utils.parse_plant_protection(json_file)
+        )
+
+    # Return the report as a response (binary)
+    headers = {
+        "Content-Disposition": "attachment; filename={}".format(report_db.name)
+    }
+
+    return Response(content=bytes(pdf.output()), media_type="application/pdf", headers=headers)
+
+
+@router.post("/{report_type}/dataset/{dataset_id}", response_model=ReportDBID)
+def create_by_data_id(
+        dataset_id: int,
+        report_type: str,
+        current_user: User = Depends(deps.get_current_user),
+        db: Session = Depends(deps.get_db)
+) -> ReportDBID:
+    """
+    Generate a report based off of a previously uploaded/queried data file.
+    [Currently returns an example PDF file regardless of input json]
+    Types: [work-book, plant-protection, irrigations, fertilisations, harvests, GlobalGAP]
+    """
+
+    # short term solution, should be in a separate model in the DB, or a part of an existing model.
+    types = {"work_book": "work-book", "plant_protection": "plant-protection", "irrigation": "irrigations", "fertilisation": "fertilisations", "harvest": "harvests", "global_gap": "GlobalGAP"}
+
+    dataset_db = crud.data.get(db=db, id=dataset_id)
+
+    if not dataset_db:
+        raise HTTPException(
+            status_code=400,
+            detail="Dataset with ID:{} does not exist.".format(dataset_id)
+        )
+
+    if report_type not in types.values():
+        raise HTTPException(
+            status_code=400,
+            detail="Report type {} isn't part of the offered types.".format(report_type)
+        )
+
+    # Create DB entry
+    report_db = crud.report.create(db=db, obj_in=ReportCreate(name=dataset_db.filename + " report.pdf", file=dataset_db.data, type=report_type))
```

Comment:
why are you saving in the `Report.file` the `Data.file`? In this way you save the same file in the DB twice, I am right? I would expect to save in the db the generated pdf file of the report.

Reply:
The report can be saved to the database instead of generating on the fly in the API, I choose this way because it's faster, for now at least.
If a huge report is requested, or one that contains a lot of data, then this method will be slower instead of caching the report and fetching it from the db when needed.
I'd make a separate feature issue of this and implement the changes in the coming weeks

Reply:
Ok. Let's create a separate issue for saving the created pdf report file to the DB.

In any case, saving `dataset.data` to `Report.file` seems a big data duplication on the DB.